internal/remotestate/backend/s3/backend.go

```diff
@@ -138,6 +139,17 @@ func (backend *Backend) Delete(ctx context.Context, backendConfig backend.Config
 		return err
 	}
 
+	if !opts.ForceBackendDelete {
```

Comment:
NIT: I'd rather pass it as an argument to the function.

Reply:
We shouldn't have conflicting sources of truth. If we're going to take it as an argument, then we shouldn't have `opts` as an argument, as that will have another copy of the same value.

Reply:
It would be great if `remotestate` didn't know anything about options at all, as stated in the book `Clean Architecture`, the called method shouldn't know anything about the properties etc. of the called method.

Reply:
I tried to implement this during the refactoring, but because we pass some authorization parameters via options, I didn't spend much time on it.

Reply:
I totally agree, but as it stands, it does. While it has access to the `opts` variable, it shouldn't receive potentially conflicting data from a different variable. 

If we want to make it agnostic, we need to refactor out all usage of `opts`, including in `NewClient`, etc. so that there isn't risk of conflicting information. 

e.g. What if I set the argument `forceBackendDelete` to true, but `opts.ForceBackendDelete` is false? It's an inconsistency that can lead to bugs.

Reply:
It's better to avoid that kind of refactor without making it all we do in the PR. We use `opts` everywhere, so we need to be careful with how we remove usage.